simpletransformers/data/url_vocab.py

```diff
@@ -14,7 +18,12 @@
 
 class UrlVocab:
     """A vocabulary of entities (articles identifiable by URL) that are recommendation candidates."""
-    def __init__(self, df_url):
+    def __init__(self, df_url,
```

Comment:
After the paper submission when things slow down, it could be worthwhile to decouple the task specific models & data manipulation logic (like this URLVocab) from the generic transformer model logic.    Perhaps we develop a separate repo for the task-specific logic that can be integrate and run with subsequent releases of `simpletransformers` without much, or any, refactoring.

Perhaps we can brainstorm a design that builds on top of simpletransformers such that we can contribute directly to the main project (PR adding generic models) rather than "hard" forking it in a way that makes it difficult to incorporate new improvements to the main repo.

As this library is so young and currently under active development, I'd expect it change (hopefully improve) quite a bit the near future

https://github.com/ThilinaRajapakse/simpletransformers/graphs/commit-activity

Reply:
Good idea, let's do that next week